### ISSUE TITLE
Implement Google Contact Sync

### ISSUE THREAD

**Issue by isthisme** [microg/android_packages_apps_GmsCore#222]
This would be a nice feature to have since the contact sync .apk from Google is proprietary.

**Comment by mar-v-in** [comment]
You should be able to sync contacts using the open-source CardDAV client DAVdroid (available on F-Droid).
Use the following details to setup the account:
- "Login with URL and user name"
- Base URL: "https://www.googleapis.com/"
- User name: Full Google account name (including @googlemail.com/gmail.com)
- Password: Your Google account password

I am not sure if all features are implemented correctly in Google's CardDAV interface. It's the same API used for Apple clients, so it shouldn't be too bad ;)

**Comment by JonnyTech** [comment]
This only seems to work with Google contacts - is there a way to sync Google calendar too?

**Comment by mar-v-in** [comment]
It should be possible to install and use the Google Calendar app

**Comment by JonnyTech** [comment]
I am using CM13 and it has a calendar app - looking at app details it is com.android.calendar version 6.0.1 - but DAVdroid has the issue connecting to the sync service, is there anything that I can check?

**Comment by mar-v-in** [comment]
There is a dedicated "Google Calendar" app in the Play Store, that's not the same app as the AOSP calendar (or derivates thereof)

**Comment by ale5000-git** [comment]
There is a Calendar Sync apk that possibly work together the AOSP calendar but I'm not sure.

**Comment by JonnyTech** [comment]
Thanks, I found a solution, use the following server url with DAVdroid:

[https://www.google.com/calendar/dav/my.email.address@google.com/events](https://www.google.com/calendar/dav/my.email.address@google.com/events)

Replace my.email.address@google.com with your full GMail address.

It syncs both my contacts and calendar from that location.

Only my primary calendar is visible but that is all that I need.

Reference: [https://github.com/gggard/AndroidCaldavSyncAdapater/wiki/Server-Compatibility-List](AndroidCaldavSyncAdapater Server Compatibility List)

**Comment by julianwi** [comment]
If somebody needs to sync a non primary calendar, look in the google calendar web interface for the Calendar ID. Press the gear top right, select settings, select calendars, then select the calendar you want to sync.
The Calendar ID should be something like d4jg8dj5ng8ek5jdhrn4udje8@group.calendar.google.com
Then use [https://www.google.com/calendar/dav/my.calendar.id@google.com/events](https://www.google.com/calendar/dav/my.calendar.id@google.com/events) as base URL in DAVdroid.
Replace my.calendar.id@google.com with your calendar id.

**Comment by Lanchon** [comment]
can google contact and/or calendar sync client be used without granting it system permissions?

**Comment by Lanchon** [comment]
gmail sync is working. i installed GoogleContactsSyncAdapter.apk as a regular app and i haven't been able to make it work.

**Comment by Lanchon** [comment]
same dance, i installed GoogleCalendarSyncAdapter.apk, enabled the calendar permission, enabled calendar access on the privacy guard popup, and cal sync is working too!

supports multiple calendars.

**Comment by coyoteltd** [comment]
I'm trying to get this working, but it keeps throwing "Couldn't find CalDAV or CardDAV service."

After some digging, it seems that Google is swatting me down with an authentication error.  But why?

**Comment by JonnyTech** [comment]
Go here: https://accounts.google.com/b/0/displayunlockcaptcha

**Comment by coyoteltd** [comment]
Thank you!  But did not fix. :(

**Comment by JonnyTech** [comment]
Have you allowed "less secure" access? https://support.google.com/accounts/answer/6010255?hl=en

**Comment by coyoteltd** [comment]
Also did not fix.  :(

**Comment by JonnyTech** [comment]
Are you using the URLs from https://github.com/microg/android_packages_apps_GmsCore/issues/222#issuecomment-261253801

Can you view your contacts at contacts.google.com

**Comment by coyoteltd** [comment]
Yes, I can view my contacts just fine.  I've tried with every URL in this thread and one I found by Googling.

I've checked my password 20 times (it's definitely right) and tried with <myemail> <myemail>@gmail.com <myemail>@googlemail.com. . .

The logs just say "Invalid Credentials" every time.

**Comment by JonnyTech** [comment]
Is there a special character in your password? Try setting it to something simple.

**Comment by anunaym14** [comment]
I refereed to the second post here to setup DAVDroid with microg but the error I get is CalDAV/CardDAV service not found!
Please help ASAP!

**Comment by JonnyTech** [comment]
Keep reading... see my reply above: https://github.com/microg/android_packages_apps_GmsCore/issues/222#issuecomment-261253801

**Comment by anunaym14** [comment]
Tried it! Same error!
I have to replace it with my email right?
I tried by replacing to anunaym14@gmail.com and got the same error!

**Comment by anunaym14** [comment]
If u want I can send you logs via telegram.
Ping me @anunaym14

**Comment by JonnyTech** [comment]
In DAVdroid: Add account >> Select "Login with URL and user name"

Base URL = https://www.google.com/calendar/dav/my.email.address@google.com/events
User name = anunaym14@gmail.com
Password = \<your-password\>

I just checked it with my account and it works.

Did you allow "less secure apps" on your google account? https://github.com/microg/android_packages_apps_GmsCore/issues/222#issuecomment-269888617

**Comment by anunaym14** [comment]
How to allow "less secure apps" on google account?
I tried again, didn't work!

**Comment by JonnyTech** [comment]
It is best to do it on a computer:

- open a web browser
- log into your Google email
- go to https://www.google.com/settings/security/lesssecureapps
- turn on "Allow less secure apps"

Now you can use non Google products, such as DAVdroid to access your account

**Comment by anunaym14** [comment]
Thank you so much..this worked!
Now, what's the difference between contact group method?
Used the method in post #2 
Pls answer..

**Comment by JonnyTech** [comment]
I do not understand your question. If it is not related to this issue then please open another.

**Comment by anunaym14** [comment]
What to select in the next step?

**Comment by anunaym14** [comment]
Thank you so much for this blazing fast support!

**Comment by matttbe** [comment]
Hello,

Thank you for the tip but note that it seems the previous URL is deprecated. I was able to use something like for my "business" account:

    https://calendar.google.com/calendar/dav/<full@email.tld>/events

But we should use one of these two URLs:

    https://apidata.googleusercontent.com/caldav/v2/<CAL_ID>/user
    https://apidata.googleusercontent.com/caldav/v2/<CAL_ID>/events

More info: https://developers.google.com/google-apps/calendar/caldav/v2/guide

**Comment by ildar** [comment]
https://github.com/microg/android_packages_apps_GmsCore/issues/222#issuecomment-268444784 is worth adding to FAQ. Properly rephrased, of course.
